### PR TITLE
handled "provider started to reconnect error"

### PR DESCRIPTION
**Pull request by kr-pankil**
## Description
I would like to know the use case of condition: "if (this.responseQueue.size > 0)",  in web3-ws-provider package.
As public RPC nodes are breaking web-socket connections frequetly, we will get the error "provider started to reconnect before response for received.". 
And after reconnect we are getting receipt of pending txs. then what is use of line "_this.responseQueue.delete(key);" 
And this type of error restart our servers or and due to for-each loop we cant handle in our code, so need to handle it from package. 
please correct me if i missed here something.

issues: #3886 and #4016

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist for 1.x:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-providers-ws/src/index.js [hunk 1]
```diff
@@ -392,8 +392,12 @@ WebsocketProvider.prototype.reconnect = function () {
 
     if (this.responseQueue.size > 0) {
         this.responseQueue.forEach(function (request, key) {
-            request.callback(errors.PendingRequestsOnReconnectingError());
-            _this.responseQueue.delete(key);
+            try {
+                request.callback(errors.PendingRequestsOnReconnectingError());
+            } catch (e) {
+                _this.responseQueue.delete(key);
```
Comment (jdevcs) [inline packages/web3-providers-ws/src/index.js:398]: Once request is sent to eth node , its moved from requestQueue to responseQueue, and with new message/response received its id is matched with responseQueue and result is sent back to user by web3 lib.

This change will not clear `responseQueue` on reconnects in normal flow(if there is no error thrown by callback), so for any requests sent to eth node web3.js will continue keeping requests in queue.

Comment (kr-pankil) [reply packages/web3-providers-ws/src/index.js:398]: so with-out this change, if get the error "provider started to reconnect ...." then i should not get the receipt of pending txs right?  as "_this.responseQueue.delete(key);" will clear response queue. but after reconnect i am getting the receipt of tx. i wonder how.

Comment (kr-pankil) [reply packages/web3-providers-ws/src/index.js:398]: any update on this ? @jdevcs

Comment (jdevcs) [reply packages/web3-providers-ws/src/index.js:398]: fixed in https://github.com/web3/web3.js/pull/5820

**Pull request closed without merging**